config/pipelines/tasks/generate-jupyterhub.yaml

```diff
@@ -0,0 +1,294 @@
+
+            if "elyra" in parsed.data["default"]:
+                return version + "-elyra"
+            else:
+                return version
+
+
+        def analyze_overlay(overlay, root):
+            print(f'Analyzed overlay \'{overlay.get("name")}\'')
+            version = python_from_pipfile(f'{root}/{overlay.get("name")}')
+            base_image = overlay.get("build", {}).get("base-image")
+
+            print(f" - Python version used:   {version}")
+            print(f" - Preferred base image:  {base_image}")
+
+            return version, base_image
+
+
+        def guess_best_match(analyzed_overlays):
+            for version, base, tag in IMAGE_LIST:
+                match = f"{base}:{tag}"
+                if version in [o[0] for o in analyzed_overlays]:
+                    return match
+                    break
+                if base in [o[1].split(":")[0] for o in analyzed_overlays]:
+                    return match
+                    break
+
+            return DEFAULT_IMAGE
+
+
+        def write_output(answer):
+            print("\nBest matching base image: " + answer)
+            with open("/mnt/baseImage", "w+") as output:
+                output.write(answer)
+
+
+        def fallback():
+            print("Missing or empty aicoe-ci.yaml, analyzing Pipfile only.")
+            version = python_from_pipfile(".")
+            answer = guess_best_match([(version, "", "")])
+            write_output(answer)
+            sys.exit(0)
+
+
+        if not os.path.isfile(".aicoe-ci.yaml"):
+            fallback()
+
+        with open(".aicoe-ci.yaml", "r") as stream:
+            conf = yaml.safe_load(stream)
+
+        if not isinstance(conf, dict):
+            fallback()
+        elif conf.get("overlays"):
+            overlays, root = (conf.get("overlays"), conf.get("overlays_dir"))
+        elif conf.get("build"):
+            overlays, root = (conf, "")
+        else:
+            fallback()
+
+        accumulator = [analyze_overlay(overlay, root) for overlay in overlays]
+        answer = guess_best_match(accumulator)
+        write_output(answer)
+
+    - name: generate
+      image: quay.io/openshift-pipeline/s2i:nightly
+      workingDir: $(workspaces.source.path)
+      volumeMounts:
+        - mountPath: /mnt
+          name: context
+      script: |
+        REPONAME=$(basename $(params.url) .git)
+        BASE_IMAGE=$(cat /mnt/baseImage)
+
+        /usr/local/bin/s2i \
+        --loglevel=0 \
+        build \
+        ./repo \
+        "$BASE_IMAGE" \
+        --as-dockerfile \
+        ./Dockerfile \
+        --env=UPGRADE_PIP_TO_LATEST=1 \
+        --env=THAMOS_RUNTIME_ENVIRONMENT="" \
+        --env=THOTH_ADVISE=0 \
+        --env=THOTH_ERROR_FALLBACK=1 \
+        --env=THOTH_DRY_RUN=1 \
+        --env=THAMOS_DEBUG=0 \
+        --env=THAMOS_VERBOSE=1 \
+        --env=THOTH_PROVENANCE_CHECK=0 \
+        --env=GIT_REPO_URL=$(params.url) \
+        --env=GIT_REPO_NAME="$REPONAME"
+  #
+  # sidecars:
```

Comment:
this has to be added to another task to feedback the log to project thoth

Reply:
Yes. Once you're back from PTO let's sync on how thoth reporting is handled and what it actually means to send images back to Thoth etc. (If it makes sense in scenarios like when the image is private and stored in cluster only, on a different cluster than where Thoth is located)

Can we keep them off for now? Let's get the initial flow without thoth reporting and I'll work on that bit right after it. WDYT?

I've also added the `send-for-image-analysis` step back (commented out)